lib/ml/feature_label.hpp

```diff
@@ -14,55 +14,29 @@
 
 #pragma once
 
-#include <cassert>
-#include <cmath>
-#include <string>
-#include <utility>
-#include <vector>
-
-#include "boost/tokenizer.hpp"
-#include "core/engine.hpp"
+#include "lib/vector.hpp"
 
 namespace husky {
 namespace lib {
 namespace ml {
-using vec_double = std::vector<double>;
 
-template <typename FT, typename LT>
-class FeatureLabelBase {
+// extends LabeledPoint, using Vector to represent features
+template <typename FeatureT, typename LabelT, bool is_sparse>
+class FeatureLabel : public LabeledPoint<Vector<FeatureT, is_sparse>, LabelT> {
```

Comment:
The suffix Impl is not so suitable because the original class LabeledPoint is already a complete class but not an interface.

How about LabeledPointHObj, where HObj is the short form of Husky object?